avm/ptn/azd/aks/main.bicep

```diff
@@ -0,0 +1,465 @@
+    primaryAgentPoolProfile: primaryAgentPoolProfile
+    dnsPrefix: dnsPrefix
+    agentPools: agentPools
+    roleAssignments: [
+      {
+        name: aksClusterRoleAssignmentName
+        principalId: principalId
+        principalType: principalType
+        roleDefinitionIdOrName: aksClusterAdminRole
+      }
+    ]
+  }
+}
+
+module containerRegistry 'br/public:avm/res/container-registry/registry:0.4.0' = {
+  name: '${uniqueString(deployment().name, location)}-container-registry'
+  params: {
+    name: containerRegistryName
+    location: location
+    tags: tags
+    publicNetworkAccess: publicNetworkAccess
+    scopeMaps: scopeMaps
+    acrSku: acrSku
+    diagnosticSettings: [
+      {
+        logCategoriesAndGroups: [
+          {
+            category: 'ContainerRegistryRepositoryEvents'
+            enabled: true
+          }
+          {
+            category: 'ContainerRegistryLoginEvents'
+            enabled: true
+          }
+        ]
+        workspaceResourceId: !empty(logAnalyticsName) ? logAnalytics.id : ''
+        metricCategories: [
+          {
+            category: 'AllMetrics'
+            enabled: true
+          }
+        ]
+      }
+    ]
+    roleAssignments: [
+      {
+        name: containerRegistryRoleName
+        principalId: managedCluster.outputs.kubeletIdentityObjectId
+        roleDefinitionIdOrName: acrPullRole
+      }
+    ]
+  }
+}
+
+module keyVault 'br/public:avm/res/key-vault/vault:0.7.1' = {
+  name: '${uniqueString(deployment().name, location)}-key-vault'
+  params: {
+    name: keyVaultName
+    accessPolicies: [
+      {
+        objectId: managedCluster.outputs.kubeletIdentityObjectId
+        permissions: {
+          secrets: ['get', 'list']
+        }
+      }
+    ]
+  }
+}
+
+// ============ //
+// Outputs      //
+// ============ //
+
+@description('The resource group the application insights components were deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The resource name of the AKS cluster.')
+output managedClusterName string = managedCluster.outputs.name
+
+@description('The Client ID of the AKS identity.')
+output managedClusterClientId string = managedCluster.outputs.kubeletIdentityClientId
+
+@description('The Object ID of the AKS identity.')
+output managedClusterObjectId string = managedCluster.outputs.kubeletIdentityObjectId
+
+@description('The resource ID of the AKS cluster.')
+output managedClusterResourceId string = managedCluster.outputs.kubeletIdentityResourceId
+
+@description('The resource name of the ACR.')
+output containerRegistryName string = containerRegistry.outputs.name
+
+@description('The login server for the container registry.')
+output containerRegistryLoginServer string = containerRegistry.outputs.loginServer
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type agentPoolType = {
+  @description('Required. The name of the agent pool.')
+  name: string?
```

Comment:
the description says it's required, but the `?` makes it optional. Which is it? :wink: